src/quera_ahs_utils/plotting.py

```diff
@@ -248,3 +244,182 @@ def plot_avg_density(densities, register, with_labels = True, custom_axes = None
     else:
         return None,ax
 
+
+
+#%%
+def visualize_UDG(positions,
+                  radii,
+                  ax=None,
+                  label_nodes=False,
+                  draw_disks=False,
+                  color_nodes=3,
+                  color_edges=0,
+                  scale=1,
+                  vertex_edge = False):
+    '''
+    Draws a unit disk graph.
+    positions - 2d positions of each vertex
+    radii     - Float. radius of each vertex. Vertices are connected if the unit disks overlap.
+                !! This is a factor of 2 from usual !!
+    draw_disks - If unit disks should be drawn.
+            True  : Draw disks for all vertices
+            False : Do not draw disks
+            list  : list of True False for each vertex
+    color_nodes - The color for each vertex.
+            int   : color assignment for all vertices.
+                0   > Black.            Indicates an inert / ground state Rydberg atom
```

Comment:
While I can agree with having an argument that accepts a type and a list of a type, this seems excessive. I think it would be best to either support a limited palette of colors that are integer accessible OR just support plain hex but not both.